conda/plugins/types.py

```diff
@@ -34,6 +38,9 @@ class CondaSubcommand(NamedTuple):
         [list[str]],  # arguments
         int | None,  # return code
     ]
```

Comment:
Will it be better to use overloading here?

```python
@dataclass
class CondaSubcommand:
    name: str
    summary: str
    action: Callable[[Namespace | tuple[str]], int | None]
    configure_parser: Callable[[ArgumentParser], None] | None = field(default=None)

    @overload
    def __init__(self, name: str, summary: str, action: Callable[[Namespace], int | None], configure_parser: Callable[[ArgumentParser], None]) -> None:
        ...

    @overload
    def __init__(self, name: str, summary: str, action: Callable[[tuple[str]], int | None], configure_parser: None = None) -> None:
        ...

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
```

Reply:
Huh, where is the `overload` decorator coming from?

Reply:
`typing.overload`

Reply:
this is good enough: https://github.com/conda/conda/pull/12814/commits/cb3efa5fe1551382d1616ce2afa55aaca70ec43b